addon/components/polaris-popover.js

```diff
@@ -90,11 +93,73 @@ export default Component.extend({
   /*
    * Internal properties.
    */
+  verticalPosition: computed('preferredPosition', function() {
+    let preferredPosition = this.get('preferredPosition');
+
+    if (preferredPosition === 'above') {
+      return preferredPosition;
+    }
+
+    if (preferredPosition === 'below') {
+      return preferredPosition;
+    }
+
+    // If set to `mostSpace`, the value will be calculated
+    // and set when the user opens the popover.
+
+    return null;
+  }),
+
   triggerStyle: computed(function() {
     return htmlSafe(`
       display: inline-block;
       overflow: inherit;
       border: none;
     `);
   }).readOnly(),
+
+  /**
+   * Checks the dropdown trigger's location on
+   * screen to determine which vertical direction
+   * has more space to open the dropdown.
+   */
+  mostVerticalSpace() {
+    let component;
+
+    // Hack to get the component's container
+    // element since this component uses `tagName: ''`
```

Comment:
Add a link here to the RFC I linked you to so we know where this came from in the future 👍

Reply:
good with me for now, ideally I think we should investigate refactoring this at some point and not using `ember-basic-dropdown`